README.md

```diff
@@ -236,46 +255,82 @@ The returned array contains objects of type `StreemCallLogEntry`, which have the
     endDate: Date?             // Session end time.
     participants: [StreemUser] // Session participants. One for an Onsite Streem, two for a two-way Streem.
     streemshotsCount: Int      // The number of Streemshots captured during the session.
+    hasMesh: Bool              // Whether the streem has a mesh with it or not.
 ```
 
 ### Displaying and Editing Streemshots
 
-Once you have fetched the call log, you may display or edit the Streemshots associated with any of the calls.
+Once you have fetched the call log, you may display or edit the artifacts associated with any of the calls.
 
-First, obtain a `StreemshotManager` for the call:
+First, obtain an `ArtifactManager` for the call. The call to `artifactManager` takes a callback which will provide the index and type of the artifact, as well as if that artifact was retrieved successfully:
 
 ```swift
-    let streemshotManager = Streem.sharedInstance.streemshotManager(forCallLogEntry: entry)
+    let artifactManager = Streem.sharedInstance.artifactManager(forCallLogEntry: entry) { [weak self] artifactType, artifactIndex, success in
+        switch artifactType {
+        case .streemshot: 
+            handleStreemshotLoading(artifactIndex, success)
+        case .recording:
+            handleRecordingLoading(artifactIndex, success)
+        case .mesh:
+            handleMeshLoading(artifactIndex, success)
+        }
+    }
 ```
 
-As soon as the `StreemshotManager` is created, it will begin to download the call's Streemshots.
+As soon as the `ArtifactManager` is created, it will begin to download the call's Streemshots.
 
-You can associate a `UIImageView` with each Streemshot. For example, this might be an image view within a UICollectionViewCell:
+Your `handleArtifactLoading` methods from above should check for success and then retrieve the image, recording, or provide a representation of the mesh. 
 
 ```swift
-    streemshotManager.register(imageView: cell.streemshotThumbnail, forStreemshotIndex: indexPath.item)
-```
+    func handleStreemshotLoading(index: Int, success: Bool) {
+        if success {
+            let image = artifactManager.streemshotImage(at: index)
+            // Do something with the image
+        } else {
+            // handle failure
+        }
+    }
 
-When the Streemshot has been downloaded, the `StreemshotManager` will set the `UIImageView`'s `image`.
+    func handleRecordingLoading(index: Int, success: Bool) {
+        if success {
+            let url = artifactManager.recordingUrl(at: index)
+            let asset = AVURLAsset(url: url, options: nil)
```

Comment:
Could mention that the artifact will be a fully downloaded, playable movie file (though probably best to not commit ourselves to providing `.mp4` specifically).

StreemNow doesn't actually make an `AVURLAsset`, but simply hands the URL to an `AVPlayerViewController()`. Up to you whether you want to mention/illustrate that option, or whether you feel that you've already suggested it sufficiently by referring to `AVURLAsset`.

Reply:
Good suggestions. I've added in both places.